contracts/true-currencies/TrueCurrencyWithPoR.sol

```diff
@@ -30,7 +28,7 @@ abstract contract TrueCurrencyWithPoR is TrueCurrency, IPoRToken {
      * @param amount The amount of tokens to mint
      */
     function _mint(address account, uint256 amount) internal virtual override {
-        if (chainReserveFeed == address(0)) {
+        if (chainReserveFeed == address(0) || !proofOfReserveEnabled) {
             super._mint(account, amount);
             return;
         }
```

Comment:
nitpick: it's probably better to keep only one canonical way to enable/disable PoR.
```suggestion
        if (!proofOfReserveEnabled) {
            super._mint(account, amount);
            return;
        }
        require(chainReserveFeed != address(0), "TODO err msg");
```

Reply:
When I first saw the suggested change I thought it was really neat! However, after thinking about it a bit more I came to the following conclusions.

1. When we realised that disabling (temporarily) PoR will be needed often and will be performed by operations people that manage minting as well as USD funds allocation across escrow accounts on a daily basis, rather than the owner account only, which is of the highest security and normally only used for upgrading the contract, we came up with an *additional*, safe method to disable temporarily PoR.
2. The original method of disabling PoR by setting `chainReserveFeed` to `0` stays in place. Still, only the owner of the contract can use it.
3. The suggested change does not eliminate any of the two available methods to disable PoR, it only causes the mint to be reverted if the two methods were used inconsistently: the owner of TUSD has not enabled PoR, yet the other method was used to enable PoR - perhaps by one of the "PoR enablers" or by the owner himself, by a mistake. In this situation I think it is right to accept authority of the owner and consider PoR as disabled, rather than consider it a situation of an inconsistent state of the contract that justifies reverting mints.

So, I think `chainReserveFeed == address(0) || !proofOfReserveEnabled` evaluated to `true` correctly means that PoR is disabled. 

I admit it doesn't feel right that the state of the contract can be somewhat inconsistent when `chainReserveFeed` is set to a non-zero address and `proofOfReserveEnabled` is `true` so I propose that:
- in `setChainReserveFeed` when the new address is `0` we also set `proofOfReserveEnabled` to `false`,
- in `_,mint` when we detect the inconsistent state we emit an event rather than revert, so it is easier to notice the situation.

What do you think?

Reply:
> In this situation I think it is right to accept authority of the owner and consider PoR as disabled, rather than consider it a situation of an inconsistent state of the contract that justifies reverting mints.

I don't have a strong opinion on this :) Your reasoning here makes sense to me, so I think we can stay with the current implementation and have two ways to pause.

> in setChainReserveFeed when the new address is 0 we also set proofOfReserveEnabled to false,
> in _,mint when we detect the inconsistent state we emit an event rather than revert, so it is easier to notice the situation.

An alternate proposal:
- `setChainReserveFeed(0)` -> `proofOfReserveEnabled = false;`
- `enableProofOfReserve() { require(chainReserveFeed != address(0)); ... }`

This way it shouldn't ever be possible to have `proofOfReserveEnabled == true` while `chainReserveFeed` is 0. How does that look?

Reply:
Ah, I see the impl has already changed to this! Looks good.

Reply:
Yep! Can you please review again and approve if you're ok with all changes?